backend/backend/MathGame/MathGameController.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace backend.MathGame.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class MathGameController : ControllerBase
+{
+    private static readonly string puzzleFilePath = Path.Combine("data", "puzzles.txt");
+    private static List<string> puzzles = new List<string>();
+    private static int puzzleIndex;
+    static MathGameController()
+    {
+        LoadPuzzlesFromFile();
+    }
+    [HttpGet("next")]
+    public ActionResult<string> GetNextPuzzle()
+    {
+        if (puzzles.Count == 0)
+        {
+            return NotFound("No puzzles available. Please check the puzzle file.");
+        }
+
+        if (puzzleIndex >= puzzles.Count)
+        {
+            puzzleIndex = 0;
+        }
+
+        string puzzle = puzzles[puzzleIndex];
+        puzzleIndex++;
+        return puzzle;
+    }
+
+    [HttpPost("solve")]
+    public ActionResult<bool> CheckAnswer([FromBody] SolvePuzzleRequest request)
+    {
+        bool isCorrect = CheckPuzzleAnswer(request.Puzzle, request.Answer);
+        return Ok(isCorrect);
+    }
+
+    private static void LoadPuzzlesFromFile()
+    {
+        try
+        {
+            if (System.IO.File.Exists(puzzleFilePath))
+            {
```

Comment:
Personally as cia patikrinciau atvirkstini atveji. Tada jei neegzistina early returnas or smth, o tada be else eiciau i happy path.
```
if (bad) {
    print "kazkas negerai"
    return
}

<skaityt faila ir pan>
```
Vykdymas identiskas, bet lengviau skaityt ir kodas neprimena HTML'o.

Reply:
Yep, agreed